### PR TITLE
MULE-12057: DataWeave should be used to encrypt payloads in idempotent-secure-hash-message-validator

### PR DESCRIPTION
**Pull request by lucianoRM**

### CODE REVIEW

File: integration/src/test/resources/core-namespace-routers-flow.xml [hunk 1]
```diff
@@ -5,13 +5,31 @@
        http://www.mulesoft.org/schema/mule/core http://www.mulesoft.org/schema/mule/core/current/mule.xsd">
 
     <flow name="IdempotentSecureHashReceiverRouter">
-        <idempotent-secure-hash-message-validator messageDigestAlgorithm="SHA-128"/>
+        <idempotent-message-validator idExpression="%dw 2.0
+            output text/plain
+            import dw::Crypto
+            ---
+            Crypto::hashWith(payload,'SHA-256')"/>
```
Comment (elrodro83) [inline integration/src/test/resources/core-namespace-routers-flow.xml]: the test was using SHA-128 before. Is that available with DW here?

File: integration/src/test/resources/org/mule/test/integration/notifications/message-processor-notification-test-flow.xml [hunk 2]
```diff
@@ -183,7 +183,11 @@
     </flow>
 
     <flow name="idempotent-secure-hash-msg-filter">
-        <idempotent-secure-hash-message-validator idExpression="#[id]" objectStore="objectStore"/>
+        <idempotent-message-validator idExpression="%dw 2.0
+        output text/plain
+        import dw::Crypto
+        ---
+        Crypto::SHA1(variables.id)" objectStore="objectStore"/>
```
Comment (elrodro83) [inline integration/src/test/resources/org/mule/test/integration/notifications/message-processor-notification-test-flow.xml]: i think the previous version was referring to the id of the event, not a variable

File: integration/src/test/resources/core-namespace-routers-flow.xml (hunk 1 above)
Comment (lucianoRM) [inline integration/src/test/resources/core-namespace-routers-flow.xml]: No, SHA-128 is not available within DW Crypto.

File: integration/src/test/resources/org/mule/test/integration/notifications/message-processor-notification-test-flow.xml (hunk 2 above)
Comment (lucianoRM) [inline integration/src/test/resources/org/mule/test/integration/notifications/message-processor-notification-test-flow.xml]: I think that there is not Id as an event attribute. It should've been a variable

File: integration/src/test/resources/core-namespace-routers-flow.xml (hunk 1 above)
Comment (lucianoRM) [inline integration/src/test/resources/core-namespace-routers-flow.xml]: Actually, SHA-128 doesn't even exist. The test was just checking that the string was properly saved.

File: integration/src/test/java/org/mule/test/config/spring/DslConstantsRoutersFlowTestCase.java [hunk 3]
```diff
@@ -30,13 +36,14 @@ public String getConfigFile() {
...
 
+  @Rule
+  public ExpectedException expectedException = none();
+
   @Test
   public void testIdempotentSecureHashReceiverRouter() throws Exception {
     Processor router = lookupMessageProcessorFromFlow("IdempotentSecureHashReceiverRouter");
-    assertThat(router.getClass().getName(),
-               equalTo("org.mule.runtime.core.internal.routing.IdempotentSecureHashMessageValidator"));
+    assertThat(router.getClass().getName(), equalTo(IdempotentMessageValidator.class));
```
Comment (elrodro83) [inline integration/src/test/java/org/mule/test/config/spring/DslConstantsRoutersFlowTestCase.java]: this may cause classloader issues

**Pull request merged**